kafka/channel/pkg/apis/messaging/v1beta1/kafka_channel_defaults.go

```diff
@@ -34,7 +34,7 @@ func (c *KafkaChannel) SetDefaults(ctx context.Context) {
 		c.Annotations = make(map[string]string)
 	}
 	if _, ok := c.Annotations[messaging.SubscribableDuckVersionAnnotation]; !ok {
-		c.Annotations[messaging.SubscribableDuckVersionAnnotation] = "v1"
```

Comment:
@aliok why was this at `v1`? :thinking: 

I also checked here: https://github.com/knative/eventing/pull/3169/files  and there (matching to IMC) it was at beta1...

Reply:
This doesn't mean that this is the KafkaChannel version.

This is the duck type version KafkaChannel supports. So, it should be v1.

Reply:
> I also checked here: https://github.com/knative/eventing/pull/3169/files and there (matching to IMC) it was at beta1...

It should've been v1 too. But, there's no difference between v1beta1 and v1 Channelable/Subscribable duck type, so, v1beta1 should also work. ...until v1beta1 support is deleted.

Reply:
Yeah, so we have to default the annotation to the version that we are currently storing. Why? Because for different serving versions we run the conversion hook, and we can adjust the annotation to match if we fiddle with the fields. For things that we are storing, however, we can't do any modifications because the conversion hook does not run. As @aliok says, since they are the same we should be safe here, but just wanted to give the reasoning for why we set the annotations differently and what the "correct" way to do that is. Hope that helps @matzew @aliok